infrastructure/general/setup-bootnode.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Set Up a Bootnode
+description: Learn how to configure and run a bootnode for Polkadot, including P2P, WS, and secure WSS connections with network key management and proxies.
+---
+
+# Set Up a Bootnode
+
+## Introduction
+
+Bootnodes are essential for helping blockchain nodes discover peers and join the network. When a node starts, it needs to find other nodes, and bootnodes provide an initial point of contact. Once connected, a node can expand its peer connections and play its role in the network, like participating as a validator. This guide will walk you through setting up a Polkadot bootnode, configuring P2P, WebSocket (WS), and secure WSS connections, and managing network keys. You’ll also learn how to test your bootnode to ensure it is running correctly and accessible to other nodes.
+
+## Prerequisites
+
+Before you start, you need to have the following prerequisites:
+
+<!-- TODO: What goes here? -->
+
+## Accessing the Bootnode
+
+To connect with other nodes in the network, bootnodes must be accessible through three key channels:
+
+- **P2P** - a direct peer-to-peer connection, set by `--listen-addr /ip4/0.0.0.0/tcp/<port>`. This is not enabled by default on non-validator nodes like archive RPC nodes
+- **P2P/WS** - a WebSocket (WS) connection, also configured via `--listen-addr`
+- **P2P/WSS** - a secure WebSocket (WSS) connection using SSL, often required for light clients. An SSL proxy is needed, as the node itself cannot handle certificates
+
+## Network Key
+
+Starting a node creates its node key in the `chains/<chain>/network/secret_ed25519` file. You can
+also create a node-key by `polkadot key generate-node-key` and use that node-key in the startup
+command line.
+
+It is essential you backup the node key, especially if it gets included in the `polkadot` binary
+because it gets hardcoded in the binary and needs to be recompiled to change.
+
+## Running the Bootnode
+
+A boot node can be run as follows:
+
+ ```
+ polkadot --chain polkadot --name dot-bootnode --listen-addr /ip4/0.0.0.0/tcp/30310 --listen-addr /ip4/0.0.0.0/tcp/30311/ws
+ ```
+
+This would have the p2p on port 30310 and p2p/ws on port 30311. For the p2p/wss port, a proxy would need to set up, a DNS name, and a corresponding certificate. The following example is for the popular nginx server and enables p2p/wss on port 30312 by adding a proxy to the p2p/ws port 30311:
```

Comment:
```suggestion
This would have the p2p on port 30310 and p2p/ws on port 30311. For the p2p/wss port, a proxy must set up a DNS name and a corresponding certificate. The following example is for the popular nginx server and enables p2p/wss on port 30312 by adding a proxy to the p2p/ws port 30311:
```